### PR TITLE
TST GH 29699 multilevel column is called after multiple appends

### LINKED ISSUE #29699: MultiIndex-Columns Integer and String: Column level 1 not accessable after a using append multiple times
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
pd.show_versions()

# Create Multi-Index
df = pd.DataFrame({'col1':[1,2,3], 'col2':[11,12,13]})
df = pd.concat([df], keys=['multi'], names=['level1'], axis=1)


for i in range(100):
    df[i, 'colA'] = 10
    a = pd.DataFrame({'col1':[1,2,3], 'col2':[11,12,13]})
    a = pd.concat([a], keys=['multi'], names=['level1'], axis=1)
    df = df.append(a, ignore_index=True)
    print(df['multi'])

```
#### Problem description

Raises an exception "KeyError: 'multi' ", although this should perfectly work, as the key "Multi" exists, e.g when printing df.columns.

The most strange thing is:
- The step in which this exception is raised depends on the Name of the key "Multi".. Sometimes 2nd loop, sometimes 6th,...
- During the first loop it works perfectly and prints the df

However it has to do something with using "integer" and "string" column-names for level1 --- if i use an integer, e.g. -1 instead of "Multi", this seems to work.

#### Expected Output
 
It prints 100 times the dataframe

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.1.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
machine          : AMD64
processor        : Intel64 Family 6 Model 142 Stepping 9, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 0.25.3
numpy            : 1.17.4
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 19.3.1
setuptools       : 39.0.1
Cython           : None
pytest           : 5.2.2
hypothesis       : None
sphinx           : 2.2.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.3
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by TAJD**
- [ ] closes #29699
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Not sure where the best place to put the test is. 

Also, this code generates a warning which doesn't seem to disappear when following the guidance of the warning i.e: set `sort=False`. As the default behaviour is meant to return a warning this is tested for.

### CODE REVIEW

File: pandas/tests/test_multilevel.py [hunk 1]
```diff
@@ -1697,6 +1697,23 @@ def test_reset_index_multiindex_columns(self):
...
+        #  GH 29699
+        df = pd.DataFrame({"col1": [1, 2, 3], "col2": [11, 12, 13]})
+        df = pd.concat([df], keys=["multi"], names=["level1"], axis=1)
+
+        # catch warning from not specifying sort
+        with tm.assert_produces_warning(RuntimeWarning):
+            for i in range(10):
+                df[i, "colA"] = 10
+                a = pd.DataFrame({"col1": [1, 2, 3], "col2": [11, 12, 13]})
+                a = pd.concat([a], keys=["multi"], names=["level1"], axis=1)
```
Comment (jreback) [inline pandas/tests/test_multilevel.py:1710]: add `sort=False` here (and remove the assert_productions_warning

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/test_multilevel.py [hunk 2]
```diff
@@ -1697,6 +1697,23 @@ def test_reset_index_multiindex_columns(self):
         result = df2.rename_axis([("c", "ii")]).reset_index(col_level=1, col_fill="C")
         tm.assert_frame_equal(result, expected)
 
+    def test_multilevel_multiple_append(self):
```
Comment (jreback) [inline pandas/tests/test_multilevel.py:1700]: can you move the test to pandas/tests/reshape/test_concat.py

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
@TAJD can you merge master and ping when green-ish

**Comment by arw2019** [comment]
@TAJD can you merge master again & resolve conflicts? Based on comments this looks good to go

**Comment by TAJD** [comment]
For some reason it doesn't quite work out when I merge and then run tests. The warning still appears. I've not got much time at the moment to work this one out - I hope I've not missed anything obvious.

**Comment by arw2019** [comment]
> For some reason it doesn't quite work out when I merge and then run tests. The warning still appears. I've not got much time at the moment to work this one out - I hope I've not missed anything obvious.

No problem! Thanks for looking into this anyways. Closing the PR for now - let us know whenever you'd like to pick this back up and we'll reopen

**Pull request closed without merging**